six/modules/c++/cphd/source/PVP.cpp

```diff
@@ -42,8 +43,25 @@ APVPType::APVPType() :
 {
 }
 
-Pvp::Pvp()
+Pvp::Pvp(bool hasAmpSF, bool hasFxN1, bool hasFxN2,
+         bool hasToaE1, bool hasToaE2, bool hasTDIonoSRP,
+         bool hasSignal)
 {
+    initialize();
+    specifyOptionalParameters(hasAmpSF, hasFxN1, hasFxN2,
+                              hasToaE1, hasToaE2, hasTDIonoSRP,
+                              hasSignal);
+}
+
+void Pvp::initialize()
+{
+    // Rest of the parameters are already initialized
+    // in PVPType constructor

```

Comment:
I'd rather have everything specified here, so the information is all in one place in the code.